src/components/Note.js

```diff
@@ -23,8 +23,10 @@ const Note = ({note}) => {
           }}>
           <Box>
             <Stack direction="row">
-              <IconButton aria-label="edit">
-                <EditIcon />
+              <IconButton 
+                onClick={() => openEditModal(note.id)}
```

Comment:
pass the whole note instead of just an id. Then you can change opentEditModal to this  
```
const openEditModal = (note) => {
    setNewForm(note);
    setOpen(true);
  }
  ```